### ISSUE TITLE
Macro-less Code Exec in MSWord

### ISSUE THREAD

**Issue by botherder** [securitywithoutborders/hardentools#37]
Any way to disable this?
https://sensepost.com/blog/2017/macro-less-code-exec-in-msword/

**Comment by obsti8383** [comment]
At least researching about 30 minutes with Google didn't result in any way to accomplish this :-(

**Comment by Atavic** [comment]
In most software firewalls you can get a prompt when an application tries to call another one.

As an example I allow only a few of safe 'foreing processes' and I get a warning like:

>An attempt to communicate a foreign process has been detected.
Target PID: 1532
Image Name: services.exe

...when something starts another process. I let it allow but I don't check the 'remeber this choice' for security reasons.

**Comment by Atavic** [comment]
In most software firewalls you can get a prompt when an application tries to call another one.
I allow only a few safe 'foreing processes' and I get a warning like:

>An attempt to communicate a foreign process has been detected.
Target PID: 1532
Image Name: services.exe

...when something starts another process as services.exe
I let it allow - if legitimate - but I don't check the 'remember this choice' for security reasons.

**Comment by emdel** [comment]
From a quick look at this problem there's an interesting post proposing a detection idea: https://www.securityforrealpeople.com/2017/10/exploiting-office-native-functionality.html

We are interested in registry keys for disabling this feature. There are many posts in forums and some good answers like:

> "I have been playing around in my XP Pro registry and have found that the DDE setting for various programs like Wordpad.exe (and I assume WinWord.exe) are in the Hkey_Classes_Root section."

> "Under Wordpad.Document.1 there is a subkey named shell with another subkey named open and below that a couple subkeys ONE of which is ddeexec."

Asap I will try to play with a Windows 7/10 VM and see what we can do.

**Comment by obsti8383** [comment]
Hi @emdel, from the forum posts that I found I had the impression that this keys only affect the opening of documents using DDE (effectively they described how to change the "open" command for files, so e.g. a new Excel document opens in an already running instance of Excel). But your quote about Wordpad makes me suspicious that there might be more to it. I'm really interested in the results of your tests.

**Comment by botherder** [comment]
Here it is:
https://gist.github.com/wdormann/732bb88d9b5dd5a66c9f1e1498f31a1b

**Comment by botherder** [comment]
I haven't managed yet to find directions to disable the same in Outlook.
Also, if anyone want to take a crack at implementing these changes, please let me know so I don't duplicate work.

**Comment by obsti8383** [comment]
I'm taking over the implementation

**Comment by emdel** [comment]
https://technet.microsoft.com/library/security/4053440

tl;dr: Microsoft is releasing this security advisory to provide information regarding security settings for Microsoft Office applications. This advisory provides guidance on what users can do to ensure that these applications are properly secured when processing Dynamic Data Exchange (DDE) fields.
